decimal.go

```diff
@@ -156,20 +156,10 @@ func NewFromBigInt(value *big.Int, exp int32) Decimal {
 	return newFromDecimal(decimal.NewFromBigInt(value, exp))
 }
 
-// optimized:
 // NewFromFloat converts a float64 to Decimal.
 //
 // NOTE: this will panic on NaN, +/-inf
 func NewFromFloat(f float64) Decimal {
-	picoFloat := f * float64(scale)
-	picoInt64 := int64(picoFloat)
-
-	// check if it's within range and is whole number
-	// integer overflow is accounted for via the `picoFloat == float64(picoInt64)` check
-	if picoInt64 >= minIntInFixed && picoInt64 <= maxIntInFixed && picoFloat == float64(picoInt64) {
-		return Decimal{fixed: picoInt64}
-	}
-
```

Comment:
Seems like this would pass the newly introduced test case and generally outperform the fallback implementation:

```suggestion
// optimized:
// NewFromFloat converts a float64 to Decimal.
//
// NOTE: this will panic on NaN, +/-inf
func NewFromFloat(f float64) Decimal {
	if math.IsNaN(f) || math.IsInf(f, 0) {
		return newFromDecimal(decimal.NewFromFloat(f))
	}

	// Convert float to string to avoid precision issues
	str := strconv.FormatFloat(f, 'f', -1, 64)
	d, err := NewFromString(str)
	if err != nil {
		return newFromDecimal(decimal.NewFromFloat(f))
	}
	return d
}
```

Reply:
Can't apply the suggestion directly on the PR ("Applying suggestions on deleted lines is currently not supported."), but I'll apply locally and update the PR description.

Reply:
Updated in e92aada